File: problems/1413-minimum-value-to-get-positive-step-by-step-sum/solution.py
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        min_sum = 0
        current_sum = 0
        for num in nums:
            current_sum += num
            min_sum = min(min_sum, current_sum)
        
        if min_sum >= 0:
            return 1
        else:
            return abs(min_sum) + 1

File: problems/1413-minimum-value-to-get-positive-step-by-step-sum/solution.java
class Solution {
    public int minStartValue(int[] nums) {
        int minSum = 0;
        int currentSum = 0;
        for (int num : nums) {
            currentSum += num;
            minSum = Math.min(minSum, currentSum);
        }
        return Math.max(1, 1 - minSum);
    }
}